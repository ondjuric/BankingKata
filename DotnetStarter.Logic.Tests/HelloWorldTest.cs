using System;
using System.Runtime.InteropServices.JavaScript;
using System.Security.Principal;
using ApprovalTests;
using ApprovalTests.Core;
using ApprovalTests.Reporters;
using Xunit;

namespace DotnetStarter.Logic.Tests
{
    [UseReporter(typeof(DiffReporter))]
    public class HelloWorldTest
    {
        [Fact]
        public void TestDeposit()
        {
            var a = new Account();
            a.SetDateGetter(() => new DateTime(2015, 12, 24));
            a.Deposit(500);
            Approvals.Verify(a.PrintStatement());
        }
        
        [Fact]
        public void TestDepositDate()
        {
            var a = new Account();
            a.SetDateGetter(() => new DateTime(2015, 12, 25));
            a.Deposit(500);
            
            Approvals.Verify(a.PrintStatement());
        }
        
        [Fact]
        public void TestDepositDateAndAmount()
        {
            var a = new Account();
            a.SetDateGetter(() => new DateTime(2015, 12, 26));
            a.Deposit(600);
            
            Approvals.Verify(a.PrintStatement());
        }
    }
    
}