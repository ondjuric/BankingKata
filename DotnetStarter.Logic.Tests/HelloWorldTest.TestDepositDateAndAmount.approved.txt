Date       Amount    Balance
26.12.2015 +600      600
